### DISCUSSION TITLE
`AuthnBackend#Credentials: Clone` breaks `oauth2::PkceCodeVerifier`

### DISCUSSION THREAD

**Discussion by Fyko** [maxcountryman/axum-login#142] [Q&A] [answered]
[`oauth2::PkceCodeVerifier`](https://docs.rs/oauth2/latest/oauth2/struct.PkceCodeVerifier.html#) intentionally does not implement `Clone` to [reinforce the integrity of the OAuth2 Process](https://docs.rs/oauth2/4.4.1/src/oauth2/types.rs.html#412-413). This doesn't allow passing the value via `fn authenticate(&self, credentials: Self::Credentials)` as `Credentials` demands `Clone`.

**Comment by maxcountryman** [comment]
Maybe you can go into more detail about the specific flow you'd like to support.

In our OAuth2 example, we've set up `Credentials` like this: https://github.com/maxcountryman/axum-login/blob/8ed2bbc8bbb945a6aa9c62983f99eb7f80765b6c/examples/oauth2/src/users.rs#L45-L48.

**Reply by maxcountryman** [reply, marked as answer]
I think this should also help: https://github.com/maxcountryman/axum-login/pull/157
